multiqc/modules/skewer/skewer.py

```diff
@@ -0,0 +1,148 @@
+#!/usr/bin/env python
+
+""" MultiQC module to parse logs from Skewer """
+
+from __future__ import print_function
+
+import os
+from collections import OrderedDict
+import logging
+import re
+from multiqc import config, BaseMultiqcModule
+
+# Initialise the logger
+log = logging.getLogger(__name__)
+
+
+class MultiqcModule(BaseMultiqcModule):
+    """ Skewer """
+
+    def __init__(self):
+        # Initialise the parent object
+        super(MultiqcModule, self).__init__(name='Skewer', anchor='skewer',
+                                            href="https://github.com/relipmoc/skewer",
+                                            info="is an adapter trimming tool specially designed for processing next-generation sequencing (NGS) paired-end sequences.")
+
+        self.sections = list()
+        self.skewer_data = dict()
+        self.skewer_readlen_dist = dict()
+
+        for f in self.find_log_files(config.sp['skewer'], filehandles=True):
+            self.parse_skewer_log(f)
+
+        headers = OrderedDict()
+        headers['pct_trimmed'] = {
+            'title': '% Trimmed',
+            'description': '% of reads trimmed',
+            'scale': 'RdYlGn-rev',
+            'max': 100,
+            'min': 0
+        }
+
+        self.general_stats_addcols(self.skewer_data, headers)
+
+        # Write parsed report data to a file
+        self.write_data_file(self.skewer_data, 'multiqc_skewer')
+        self.write_data_file(self.skewer_readlen_dist, 'multiqc_skewer_readlen_dist')
+
+        # set the value 0 for every x where a given sample doens't have a value
+        all_x_values = []
+        for s_name in self.skewer_readlen_dist:
+            for xval in self.skewer_readlen_dist[s_name]:
+                all_x_values.append(xval)
+
+        for s_name in self.skewer_readlen_dist:
+            for xval in all_x_values:
+                if not xval in self.skewer_readlen_dist[s_name]:
+                    self.skewer_readlen_dist[s_name][xval] = 0.0
+
+            # After adding new elements, the ordereddict needs to be re-sorted
+            items = self.skewer_readlen_dist[s_name]
+            self.skewer_readlen_dist[s_name] = OrderedDict(sorted(items.iteritems(), key=lambda x: int(x[0])))
+
+        # add the histogram to the report
+        self.add_readlen_dist_plot()
+
+        if len(self.skewer_data) == 0:
+            log.debug("Could not find any data in {}".format(config.analysis_dir))
+            raise UserWarning
+
+        log.info("Found {} reports".format(len(self.skewer_data)))
+
+    def add_readlen_dist_plot(self):
+        pconfig = {
+            'id': 'skewer_read_length_histogram',
+            'title': 'Read Length Distribution',
+            'categories': True,
+            'ylab': '% of Reads',
+            'xlab': 'Read Length',
+            'ymax': 100,
```

Comment:
Would it make sense to have the x axis start at 0? I only say this because most of the other plots (eg. FastQC) do this. Might look weird here I guess.